Spigot-API-Patches/0236-Add-worldborder-events.patch

```diff
@@ -0,0 +1,375 @@
++     * @return center as Location. Y will always be 0.0 as it is irrelevant.
++     */
++    @NotNull
++    public Location getNewLocation() {
++        return newLocation;
++    }
++
++    /**
++     * Check if this event is cancelled
++     * @return true if cancelled, false if not
++     */
++    @Override
++    public boolean isCancelled() {
++        return cancelled;
++    }
++
++    /**
++     * Set this event as cancelled
++     * @param cancel true if you wish to cancel this event
++     */
++    @Override
++    public void setCancelled(boolean cancel) {
++        this.cancelled=cancel;
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/event/world/WorldBorderFinishChangeEvent.java b/src/main/java/io/papermc/paper/event/world/WorldBorderFinishChangeEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..35940facd61186359eba4c7eae4c3acd857acb38
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/world/WorldBorderFinishChangeEvent.java
+@@ -0,0 +1,43 @@
++package io.papermc.paper.event.world;
++
++import org.bukkit.World;
++import org.bukkit.WorldBorder;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.world.WorldEvent;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when the World Border finishes a transition. **Not** called on a static change (when the delay in {@code /worldborder set} is specified as {@code 0})
++ */
++public class WorldBorderFinishChangeEvent extends WorldEvent {
++
++    private static final HandlerList HANDLERS = new HandlerList();
++
++    private final WorldBorder border;
++
++    public WorldBorderFinishChangeEvent(@NotNull World world, @NotNull WorldBorder border) {
++        super(world);
++        this.border=border;
++    }
++
++    @NotNull
++    @Override
++    public HandlerList getHandlers() {
++        return HANDLERS;
++    }
++    @NotNull
++    public static HandlerList getHandlerList() {
++        return HANDLERS;
++    }
++
++    /**
++     * Get the world border
++     * @return WorldBorder
++     */
++    @NotNull
++    public WorldBorder getWorldBorder() {
++        return border;
++    }
++
++
++}
+diff --git a/src/main/java/io/papermc/paper/event/world/WorldBorderStartChangeEvent.java b/src/main/java/io/papermc/paper/event/world/WorldBorderStartChangeEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..ef000ccd555a2c2ffc859653e08e5780a2443332
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/world/WorldBorderStartChangeEvent.java
+@@ -0,0 +1,59 @@
++package io.papermc.paper.event.world;
++
++import org.bukkit.World;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.world.WorldEvent;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when the World Border starts a transition. **Not** called on a static change (when the delay in {@code /worldborder set} is specified as {@code 0})
++ */
++public class WorldBorderStartChangeEvent extends WorldEvent {
```

Comment:
Why isn't this a generic border event? Why isn't this just a dynamic border event?